Assets/Plugins/Mediabox/Source/GameKit/GameManager/PauseHandler.cs

```diff
@@ -2,30 +2,58 @@
 
 namespace Mediabox.GameKit.GameManager {
     public class PauseHandler {
-        float timeScaleBeforePause;
-        float volumeBeforePause;
-        bool isPaused;
+        float timeScaleBeforePause = 1;
+        float volumeBeforePause = 1;
+        bool isPaused = false;
+
+        public delegate void OnPause(bool pausing);
+        public event OnPause onPause;
+        
         
         public void Pause() {
-            if (!this.isPaused) {
-                this.timeScaleBeforePause = Time.timeScale;
-                this.volumeBeforePause = AudioListener.volume;
-                this.isPaused = true;
-            } else {
+            
+            
+            if (this.isPaused){
                 Debug.LogWarning("Pause() has been called multiple times. To avoid unintended behavior, it is recommended to ensure calling Unpause() before Pausing again.");
             }
-            Time.timeScale = 0f;
+            
+            this.isPaused = true;
+            this.volumeBeforePause = AudioListener.volume;
             AudioListener.volume = 0f;
+            
+            if (onPause != null){
+                if (onPause.GetInvocationList().Length > 0){
+                    onPause.Invoke(true);
+                    return;
+                }
+            }
+            
+            this.timeScaleBeforePause = Time.timeScale;
+            Time.timeScale = 0f;
+           
         }
 
         public void Unpause() {
-            if (this.isPaused) {
-                Time.timeScale = this.timeScaleBeforePause;
-                AudioListener.volume = this.volumeBeforePause;
-                this.isPaused = false;
-            } else {
+
+            if (!this.isPaused){
                 Debug.LogWarning("Unpause() has been called multiple times. To avoid unintended behavior, it is recommended to ensure calling Pause() before Unpausing again.");
             }
+            
+            this.isPaused = false;
+            AudioListener.volume = this.volumeBeforePause;
+            if (onPause != null){
+                if (onPause.GetInvocationList().Length > 0){
+                    onPause.Invoke(false);
+                    return;
+                }
+            }
+            Time.timeScale = this.timeScaleBeforePause;
+        }
+
+        public void ResetPauseValues(){
+            onPause = null;
+            timeScaleBeforePause = 1;
+            volumeBeforePause = 1;
         }
```

Comment:
Ah! now I remember why I wrote it like this. Not having Daniel's code, I couldn't tell if UnPause was called before or after the game was loaded. If I set isPause to false before the unpause it would cause a double-unpause error. 
In addition I only set the timeScaleBeforePause and volumeBeforePause instead of just resetting the audioListener.volume and Time.time, because if I had reset the audioListener directly at this time, it could spill playing music from the game into the main app.